### ISSUE TITLE
Fix the 'app-meta!$data' stealth module hack

### ISSUE THREAD

**Issue by dchambers** [BladeRunnerJS/brjs#1477]
The [SystemJS Spike](https://github.com/BladeRunnerJS/brjs/issues/1389) was hampered by the `app-meta!$data` stealth module. Now that the server also needs awareness of all dependencies, we can no longer get away with this hack.

**Comment by dchambers** [comment]
This was fixed by 718bfed.

**Comment by andyberry88** [comment]
Commit 718bfed looks good. Given this is in the `NPM Support` branch and theres no PR for this in to master can this be closed?

**Comment by dchambers** [comment]
I'm not sure. Given that it's a long running branch, I think it can be moved to `4 - ReadyForTest`  as normal, so that we can sign off the various bits of functionality as we fix them. Obviously, there would need to be some regression testing done at the point we merge all of this into master since changes in master may break existing functionality, but then you could say the same about functionality already in master being affected by commits being merged into master. That's what the automated tests are for.

**Comment by andyberry88** [comment]
:+1: Ok so no PRs for the NPM support branch but review + QA as normal for commits in that branch.

**Comment by dchambers** [comment]
Ah, good point, maybe I should be doing pull-requests. That will definitely make sense as more of us work on this branch. Maybe I should start from this point on?

**Comment by andyberry88** [comment]
> maybe I should be doing pull-requests

This might make it easier to QA/review as it progresses but could add extra overhead for you, particularly since this is just a branch where you're the only person that's likely to be committing to it. Would it be better to add all issues that are fixed by the branch to the `NPM Support` milestone and QA/review them all once the branch is merged in a single PR?

**Comment by thecapdan** [comment]
@dchambers were you going to open a PR for this change?

**Comment by dchambers** [comment]
@thecapdan, unfortunately it's too late now, it's already merged into the `new-systemjs-spike` feature branch. I was more agreeing with @andyberry88 that all future changes to this branch should be done in sub-branches, allowing us to create pull-requests.
